compiler/rustc_smir/src/rustc_smir/context.rs

```diff
@@ -225,6 +258,29 @@ impl<'tcx> Context for TablesWrapper<'tcx> {
         tables.tcx.is_foreign_item(tables[item])
     }
 
+    fn foreign_item_kind(&self, def: ForeignDef) -> Option<ForeignItemKind> {
+        let (def_id, hir_kind) = {
+            let tables = self.0.borrow();
+            let def_id = tables[def.def_id()];
+            let hir_kind = tables
+                .tcx
+                .hir()
+                .expect_foreign_item(rustc_hir::OwnerId { def_id: def_id.as_local()? })
+                .kind;
```

Comment:
You could check the `DefKind` (via `def_kind`). That works for other crates, too.